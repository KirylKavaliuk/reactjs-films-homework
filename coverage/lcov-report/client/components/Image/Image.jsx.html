<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for client/components/Image/Image.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\index.html">All files</a> / <a href="index.html">client/components/Image</a> Image.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>8/8</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>8/8</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import PropTypes from 'prop-types';
&nbsp;
import withConditionalRendering from 'utils/rendering';
&nbsp;
import classNames from 'classnames';
&nbsp;
import styles from './Image.scss';
&nbsp;
const Image = ({
  db,
  src,
  width,
  className,
}) =&gt; {
  const imageClasses = classNames(styles.image, className);
  let transformedSrc = src;
&nbsp;
  if (db) {
    transformedSrc = `https://image.tmdb.org/t/p/w${width}/${src}`;
  }
&nbsp;
  return (&lt;img className={ imageClasses }src={ transformedSrc }/&gt;);
};
&nbsp;
Image.defaultProps = {
  db: false,
  src: '',
  width: 500,
  className: '',
};
&nbsp;
Image.propTypes = {
  db: PropTypes.bool,
  src: PropTypes.string,
  width: PropTypes.number,
  className: PropTypes.string,
};
&nbsp;
export default withConditionalRendering(Image, 'src');
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Feb 01 2019 11:31:09 GMT+0300 (GMT+03:00)
</div>
</div>
<script src="..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\sorter.js"></script>
</body>
</html>
