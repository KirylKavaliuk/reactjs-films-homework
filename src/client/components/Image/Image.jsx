import React from 'react';
import PropTypes from 'prop-types';

import withConditionalRendering from 'utils/rendering';

import classNames from 'classnames';

import styles from './Image.scss';

const Image = ({
  db, src, width, className,
}) => {
  const imageClasses = classNames(styles.image, className);
  let transformedSrc = src;

  if (db) {
    transformedSrc = `https://image.tmdb.org/t/p/w${width}/${src}`;
  }

  return (
    <img className={ imageClasses } src={ transformedSrc }/>
  );
};

Image.defaultProps = {
  db: false,
  src: '',
  width: 500,
  className: '',
};

Image.propTypes = {
  db: PropTypes.bool,
  src: PropTypes.string,
  width: PropTypes.number,
  className: PropTypes.string,
};

export default withConditionalRendering(Image, 'src');
