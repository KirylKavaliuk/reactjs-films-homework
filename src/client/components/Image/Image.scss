@import 'styles/palette';

.image {
  width: 100%;
  background: $LIGHT_GREY;
  position: relative;
  border: none;
}
