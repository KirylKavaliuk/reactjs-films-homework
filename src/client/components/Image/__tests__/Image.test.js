import React from 'react';
import renderer from 'react-test-renderer';
import Image from '../Image';

it('renders image uncorrectly', () => {
  const tree = renderer.create(<Image/>).toJSON();

  expect(tree).toBeNull();
  expect(tree).toMatchSnapshot();
});

it('renders image correctly', () => {
  const tree = renderer.create(<Image db src='/image-ib.jpg'/>).toJSON();

  expect(tree).not.toBeNull();
  expect(tree).toMatchSnapshot();
});

it('renders image correctly', () => {
  const tree = renderer.create(
    <Image
      src='images/image-ib.jpg'
      className='Component__class-name__hash'
    />,
  ).toJSON();

  expect(tree).not.toBeNull();
  expect(tree).toMatchSnapshot();
});
