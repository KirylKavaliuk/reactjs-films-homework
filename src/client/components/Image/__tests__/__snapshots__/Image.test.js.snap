// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Image/> 1`] = `null`;

exports[`<Image/> 2`] = `
<img
  className="Image__image___2zFQr"
  src="https://image.tmdb.org/t/p/w500//image-ib.jpg"
/>
`;

exports[`<Image/> 3`] = `
<img
  className="Image__image___2zFQr Component__class-name__hash"
  src="images/image-ib.jpg"
/>
`;
